### ISSUE TITLE
Smooth analog potentiometer readings

### ISSUE THREAD

**Issue by neilenns** [MobiFlight/MobiFlight-FirmwareSource#50]
Copied from https://github.com/MobiFlight/MobiFlight-Connector/issues/519

> **Is your feature request related to a problem? Please describe.**
> I have encountered some electrical noise coming from 7 segment display modules, which cause jitter problems in a potentiometer. Suggestions received have reduced the problem but not eliminated it. It would be good to have a way to smoothen potentiometer readings through software.
> 
> **Describe the solution you'd like**
> Implement a 10 reading rolling average in the arduino firmware with a soft switch in MF Connector analog input configuration, that can be used to turn this feature on and off. When on, the analog pot reading sent from the board will be the rolling average of the last 10 readings. This will effectively smoothen the readings and eliminate jitter. This technique has been successfully used to smoothen the operation of a yoke with three analog pots connected to a pro micro.
> 
> **Describe alternatives you've considered**
> I have tried to isolate the noise problem in hardware and jitter has been successfully reduced, but not eliminated.
> 
> **Additional context**
> Potentiometers can present jitter due to various reasons. This will provide a solution in software to alleviate this problem.
> I suggest that instead of calculating a real rolling average, which can take too much computational power, adding 10 readings and dividing by 10 in each read cycle, Start with an assumed average of zero and an array of ten readings (all zero). In each cycle: increment the reading counter index (0 to 9), add the current reading divided by 10 and substract the counter indexed reading divided by 10. Finally, store the current reading in the counter indexed position (replacing the value just substracted).
> Any error at start disappears within 10 cycles.

**Comment by JaimeLeon2** [comment]
BUG REPORT on the mega firmware.  It was working very well, but just to test I reloaded the mfmc file for the board.  The upload to the board appears to go well, but after exiting the MF Modules window, I was having problems and going back to MF Modules now showed the board is missing about half the configuration.  I repeated this three times with the same result.

**Comment by JaimeLeon2** [comment]
INITIAL TEST REPORT.
Hello, I was able to load and run the test version of the mega firmware with the analog input smoothing function.  Wow! The pot now is working rock solid even with the 7 segment displays on.  Almost zero jitter, maybe a 0.2 % twitch now and then, but 99% of the time just solid without any twitches.  Thank you for such a fantastic job in resolving the pot instability. 😃 👍 
SO THE SMOOTHING WORKS WONDERFULLY.

**Comment by elral** [comment]
I just tested it also. I can confirm that on the promicro after an upload of the firmware the config does not show up. After closing the connector and a restart the config is shown and everything works. I tried to do a test with the origianl firmware and after upload the config is also not shown. A restart of the connector does not help as it seems that the promicro does not run with all devices activated.
On the mega the upload works as aspected, the config does show up immediately after upload the firmware. It might be thgat there is an disconnect and reconnect after the upload, this would mean that the mega is resettet. But I am not sure.
